test/metabase/api/action_test.clj

```diff
@@ -205,3 +252,56 @@
           (is (= "Failed to fetch Table 2,147,483,647: Table does not exist, or belongs to a different Database."
                  (:message (mt/user-http-request :crowberto :post 404 action
                                                  (assoc-in request-body [:query :source-table] Integer/MAX_VALUE))))))))))
+
+(deftest bulk-create-happy-path-test
+  (testing "POST /api/action/bulk/create/:table-id"
+    (mt/test-drivers (mt/normal-drivers-with-feature :actions)
+      (actions.test-util/with-actions-test-data-and-actions-enabled
+        (is (= 75
+               (categories-row-count)))
+        (is (= {:created-rows [{(format-field-name :id) 76, (format-field-name :name) "NEW_A"}
+                               {(format-field-name :id) 77, (format-field-name :name) "NEW_B"}]}
+               (mt/user-http-request :crowberto :post 200
+                                     (format "action/bulk/create/%d" (mt/id :categories))
+                                     [{(format-field-name :name) "NEW_A"}
+                                      {(format-field-name :name) "NEW_B"}])))
+        (is (= [[76 "NEW_A"]
+                [77 "NEW_B"]]
+               (mt/rows (mt/run-mbql-query categories {:filter   [:starts-with $name "NEW"]
+                                                       :order-by [[:asc $id]]}))))
+        (is (= 77
+               (categories-row-count)))))))
+
+(deftest bulk-create-failure-test
+  (testing "POST /api/action/bulk/create/:table-id"
+    (mt/test-drivers (mt/normal-drivers-with-feature :actions)
+      (actions.test-util/with-actions-test-data-and-actions-enabled
+        (testing "error in some of the rows in request body"
+          (is (= 75
+                 (categories-row-count)))
+          (testing "Should report indices of bad rows"
+            (is (schema= {:errors [(s/one {:index (s/eq 1)
+                                           :error (s/constrained
+                                                   s/Str
+                                                   (case driver/*driver*
+                                                     :h2       #(str/starts-with? % "NULL not allowed for column \"NAME\"")
+                                                     :postgres #(str/starts-with? % "ERROR: null value in column \"name\"")))}
```

Comment:
This error shape isn't exactly what we wanted in the tech design doc but seemed close enough for an MVP. We have the row `:index` like we wanted but column errors are not keyed by the column name -- it seemed hard to do and this PR had enough going on already. We can add that stuff as a follow-on I think.